crates/bevy_ecs/src/component.rs

```diff
@@ -201,11 +203,23 @@ pub enum StorageType {
     SparseSet,
 }
 
+/// The type used for [`Component`] lifecycle hooks such as `on_add`, `on_insert` or `on_remove`
+pub type ComponentHook = for<'w> fn(DeferredWorld<'w>, Entity, ComponentId);
```

Comment:
If this was changed to `&mut DeferredWorld` you avoid needing to add `mut` in the hook closure definition, and can avoid repeated instantiation of `DeferredWorld` when triggering the hooks.

Reply:
The presence of `mut world` was not something I had considered but there is no cost in constructing `DeferredWorld` as it is just a wrapped world pointer.

Reply:
Yeah I doubt it has any performance implications, moreso just the ergonomics. Conceptually as well, it matches how things like `resource_scope` work (providing a world reference). Obviously `DeferredWorld` _is_ a reference, but it would help with first-glance clarity IMO.